File: .agents/skills/alipay-payment-integration/references/code-examples/php/10_AI收/A2MPaymentDemo.php
<?php
/**
 * A2M 智能收产品接入示例 - PHP 版本
 * 
 * 本文件演示完整的智能收产品接入流程：
 * 1. 返回 402 Payment-Needed Header
 * 2. 验证 Payment-Proof 支付凭证
 * 3. 发送履约回执确认
 * 4. 返回资源内容
 * 
 * 依赖安装：
 * composer require alipay/alipay-sdk
 */

// ==================== 配置信息（实际使用时请从配置中心读取）====================
define('ALIPAY_CONFIG', [
    'appId' => '2026000123456789',
    'privateKey' => 'MIIEvQIBADANBgkq...', // 请填写您的应用私钥（PKCS#1 格式）
    'alipayPublicKey' => 'MIIBIjANBgkq...', // 请填写您的支付宝公钥
    'gateway' => 'https://openapi.alipay.com/gateway.do',
    'sellerId' => '2088123456789012', // 商户 ID（2088 格式）
    'serviceId' => 'service_ai_content_001', // 商户服务 ID
    'merchantPrivateKey' => 'MIIEvQIBADANBgkq...' // 请填写您的应用私钥（用于商家签名）
]);

define('RESOURCE_CONFIG', [
    'path' => '/demo/a2m/resource',
    'goodsName' => 'AI 生成内容服务'
]);

/**
 * 初始化 SDK
 */
function initAlipayClient() {
    require_once 'aop/AopClient.php';
    require_once 'aop/AlipayConfig.php';
    
    $config = new AlipayConfig();
    $config->setServerUrl(ALIPAY_CONFIG['gateway']);
    $config->setAppId(ALIPAY_CONFIG['appId']);
    $config->setPrivateKey(ALIPAY_CONFIG['privateKey']);
    $config->setFormat('json');
    $config->setAlipayPublicKey(ALIPAY_CONFIG['alipayPublicKey']);
    $config->setCharset('UTF-8');
    $config->setSignType('RSA2');
    
    return new AopClient($config);
}

/**
 * 格式化支付宝时间戳：yyyy-MM-dd HH:mm:ss
 */
function formatAlipayTimestamp($timestamp = null) {
    if ($timestamp === null) {
        $timestamp = time();
    }
    return date('Y-m-d H:i:s', $timestamp);
}

/**
 * 生成商家签名（seller_signature）
 */
function generateSellerSignature($params, $privateKey) {
    // 1. 按 key 字典序排序
    ksort($params);
    
    // 2. 拼接签名内容
    $signContent = [];
    foreach ($params as $key => $value) {
        if ($value !== null && $value !== '') {
            $signContent[] = "{$key}={$value}";
        }
    }
    $signString = implode('&', $signContent);
    
    // 3. RSA2 签名
    $privateKeyResource = openssl_pkey_get_private($privateKey);
    openssl_sign($signString, $signature, $privateKeyResource, OPENSSL_ALGO_SHA256);
    
    return base64_encode($signature);
}

/**
 * Base64URL 编码
 */
function base64UrlEncode($data) {
    return rtrim(strtr(base64_encode($data), '+/', '-_'), '=');
}

/**
 * Base64URL 解码
 */
function base64UrlDecode($data) {
    $padding = strlen($data) % 4;
    if ($padding) {
        $data .= str_repeat('=', 4 - $padding);
    }
    return base64_decode(strtr($data, '-_', '+/'));
}

/**
 * 发送 JSON 响应
 */
function jsonResponse($data, $statusCode = 200, $headers = []) {
    http_response_code($statusCode);
    header('Content-Type: application/json; charset=utf-8');
    
    foreach ($headers as $key => $value) {
        header("{$key}: {$value}");
    }
    
    echo json_encode($data, JSON_UNESCAPED_UNICODE);
    exit;
}

// ==================== 智能收产品接入示例接口 ====================

/**
 * 智能收产品统一接口
 * 
 * 完整流程演示：
 * 1. 不带 Payment-Proof Header：返回 HTTP 402 + Payment-Needed Header
 * 2. 带 Payment-Proof Header：验证支付 → 自动履约 → 返回资源
 */
function handleResourceRequest() {
    // 获取 Payment-Proof Header
    $paymentProof = isset($_SERVER['HTTP_PAYMENT_PROOF']) ? $_SERVER['HTTP_PAYMENT_PROOF'] : null;
    
    // 场景 1：用户未支付，返回 402 + Payment-Needed Header
    if (empty($paymentProof) || trim($paymentProof) === '') {
        createPaymentRequiredResponse();
        return;
    }
    
    // 场景 2：用户已支付，验证 Payment-Proof 并返回资源
    verifyPaymentAndDeliverResource($paymentProof);
}

/**
 * 创建 402 支付请求响应
 */
function createPaymentRequiredResponse() {
    try {
        // 1. 构造订单信息
        $outTradeNo = 'ORDER_' . time() . mt_rand(1000, 9999);
        $amount = '0.01'; // 单位：元
        $currency = 'CNY';
        $resourceId = RESOURCE_CONFIG['path'];
        $goodsName = RESOURCE_CONFIG['goodsName'];
        
        // 2. 计算支付截止时间（30 分钟后）
        $payBefore = date('c', time() + 30 * 60); // ISO 8601 格式
        
        // 3. 生成商家签名
        $sellerSignature = generateSellerSignature([
            'amount' => $amount,
            'currency' => $currency,
            'goods_name' => $goodsName,
            'out_trade_no' => $outTradeNo,
            'pay_before' => $payBefore,
            'resource_id' => $resourceId,
            'seller_id' => ALIPAY_CONFIG['sellerId'],
            'service_id' => ALIPAY_CONFIG['serviceId']
        ], ALIPAY_CONFIG['merchantPrivateKey']);
        
        // 4. 构造 Payment-Needed Header 内容
        $paymentNeeded = [
            'protocol' => [
                'out_trade_no' => $outTradeNo,
                'amount' => $amount,
                'currency' => $currency,
                'resource_id' => $resourceId,
                'pay_before' => $payBefore,
                'seller_signature' => $sellerSignature,
                'seller_sign_type' => 'RSA2',
                'seller_unique_id' => ALIPAY_CONFIG['sellerId']
            ],
            'method' => [
                'seller_name' => '测试商户',
                'seller_id' => ALIPAY_CONFIG['sellerId'],
                'seller_app_id' => ALIPAY_CONFIG['appId'],
                'goods_name' => $goodsName,
                'seller_unique_id_key' => 'seller_id',
                'service_id' => ALIPAY_CONFIG['serviceId']
            ]
        ];
        
        // 5. Base64URL 编码
        $paymentNeededEncoded = base64UrlEncode(json_encode($paymentNeeded, JSON_UNESCAPED_UNICODE));
        
        // 6. 构造 402 响应
        $response = [
            'code' => 'Payment-Needed',
            'message' => '需要支付',
            'out_trade_no' => $outTradeNo,
            'amount' => $amount,
            'currency' => $currency,
            'goods_name' => $goodsName
        ];
        
        error_log("创建支付订单成功：outTradeNo={$outTradeNo}, amount={$amount}");
        
        jsonResponse($response, 402, ['Payment-Needed' => $paymentNeededEncoded]);
        
    } catch (Exception $e) {
        error_log('创建订单失败：' . $e->getMessage());
        jsonResponse([
            'code' => 'CREATE_ORDER_ERROR',
            'message' => '创建订单失败：' . $e->getMessage()
        ], 500);
    }
}

/**
 * 验证支付凭证并交付资源
 */
function verifyPaymentAndDeliverResource($paymentProof) {
    try {
        // 1. 从 Payment-Proof 中解析订单信息
        $decodedProof = base64UrlDecode($paymentProof);
        $proofJson = json_decode($decodedProof, true);
        
        if (json_last_error() !== JSON_ERROR_NONE) {
            throw new Exception('Payment-Proof 解析失败：' . json_last_error_msg());
        }
        
        // 从 protocol 层获取 payment_proof 和 trade_no
        $paymentProofValue = null;
        $tradeNo = null;
        $clientSession = null;
        
        if (isset($proofJson['protocol'])) {
            $paymentProofValue = $proofJson['protocol']['payment_proof'] ?? null;
            $tradeNo = $proofJson['protocol']['trade_no'] ?? null;
        }
        
        // 从 method 层获取 client_session
        if (isset($proofJson['method'])) {
            $clientSession = $proofJson['method']['client_session'] ?? null;
        }
        
        // 校验必要字段
        if (empty($paymentProofValue)) {
            jsonResponse([
                'code' => 'INVALID_PAYMENT_PROOF_FORMAT',
                'message' => 'Payment-Proof 格式错误：缺少 payment_proof'
            ], 400);
        }
        
        if (empty($tradeNo)) {
            jsonResponse([
                'code' => 'INVALID_PAYMENT_PROOF_FORMAT',
                'message' => 'Payment-Proof 格式错误：缺少 trade_no'
            ], 400);
        }
        
        // 2. 调用支付宝 API 验证支付凭证
        $alipayClient = initAlipayClient();
        $request = new \AlipayAipayAgentPaymentVerifyRequest();
        
        $model = [
            'payment_proof' => $paymentProofValue,
            'trade_no' => $tradeNo,
            'client_session' => $clientSession
        ];
        
        $request->setBizContent(json_encode($model, JSON_UNESCAPED_UNICODE));
        $responseResult = $alipayClient->execute($request);
        $responseApiName = str_replace('.', '_', $request->getApiMethodName()) . '_response';
        $verifyResponse = $responseResult->$responseApiName;
        
        // 3. 验证失败，返回错误
        if (empty($verifyResponse->code) || $verifyResponse->code !== '10000') {
            error_log('支付凭证验证失败：' . ($verifyResponse->sub_msg ?? 'Unknown error'));
            jsonResponse([
                'code' => $verifyResponse->sub_code ?? 'VERIFY_FAILED',
                'message' => $verifyResponse->sub_msg ?? '支付凭证验证失败'
            ], 400);
        }
        
        // 4. 验证成功，获取订单信息
        $verifyTradeNo = $verifyResponse->trade_no ?? null;
        $verifyOutTradeNo = $verifyResponse->out_trade_no ?? null;
        $resourceId = $verifyResponse->resource_id ?? null;
        $active = $verifyResponse->active ?? null;
        
        error_log("支付凭证验证成功：tradeNo={$verifyTradeNo}, outTradeNo={$verifyOutTradeNo}");
        
        // 5. 校验凭证有效性（active=true 表示凭证有效）
        if ($active !== true) {
            error_log("支付凭证无效或已过期：outTradeNo={$verifyOutTradeNo}");
            jsonResponse([
                'code' => 'INVALID_PAYMENT_PROOF',
                'message' => '支付凭证无效或已过期'
            ], 400);
        }
        
        // 6. 【TODO】查询订单是否存在（以数据库为准）
        // $order = $orderRepository->findByOutTradeNo($verifyOutTradeNo);
        // if (!$order) {
        //     jsonResponse([
        //         'code' => 'ORDER_NOT_FOUND',
        //         'message' => '订单不存在'
        //     ], 404);
        // }
        
        // 7. 【TODO】资源防串校验
        // if ($resourceId !== $order->getResourceId()) {
        //     jsonResponse([
        //         'code' => 'RESOURCE_ID_MISMATCH',
        //         'message' => '资源 ID 不匹配，可能存在资源串改风险'
        //     ], 403);
        // }
        
        // 8. 【TODO】履约防重放校验（数据库幂等控制）
        // if ($order->getFulfillStatus() === 'FULFILLED') {
        //     jsonResponse([
        //         'code' => 'ALREADY_FULFILLED',
        //         'message' => '订单已履约，不重复提供',
        //         'already_fulfilled' => true
        //     ], 200);
        // }
        
        // 9. 执行业务逻辑，生成资源内容
        $serviceResult = generateServiceResource($resourceId);
        
        // 10. 【TODO】履约记录落库（用于审计/售后/对账）
        // $fulfillmentRecordRepository->save([...]);
        
        // 11. 【TODO】更新订单状态
        // $orderRepository->update($verifyOutTradeNo, [...]);
        
        error_log("履约成功：outTradeNo={$verifyOutTradeNo}, tradeNo={$verifyTradeNo}");
        
        // 12. 发送履约确认到支付宝
        sendFulfillmentConfirm($verifyTradeNo);
        
        // 13. 构造 Payment-Validation Header
        $paymentValidation = [
            'trade_no' => $verifyTradeNo,
            'out_trade_no' => $verifyOutTradeNo,
            'validated' => true,
            'resource_id' => $resourceId
        ];
        
        $paymentValidationEncoded = base64UrlEncode(json_encode($paymentValidation, JSON_UNESCAPED_UNICODE));
        
        // 14. 返回资源内容
        jsonResponse([
            'resource_id' => $resourceId,
            'content' => $serviceResult,
            'trade_no' => $verifyTradeNo,
            'out_trade_no' => $verifyOutTradeNo,
            'already_fulfilled' => false
        ], 200, ['Payment-Validation' => $paymentValidationEncoded]);
        
    } catch (Exception $e) {
        error_log('支付凭证验证异常：' . $e->getMessage());
        jsonResponse([
            'code' => 'VERIFY_FAILED',
            'message' => '支付凭证验证失败：' . $e->getMessage()
        ], 500);
    }
}

/**
 * 生成服务资源内容
 */
function generateServiceResource($resourceId) {
    return json_encode([
        'status' => 'success',
        'service_type' => 'AI_CONTENT_GENERATION',
        'resource_id' => $resourceId,
        'content' => '这是 AI 生成的内容示例，可根据实际业务替换为任意数字服务内容',
        'generated_at' => date('c')
    ], JSON_UNESCAPED_UNICODE);
}

/**
 * 发送履约确认
 */
function sendFulfillmentConfirm($tradeNo) {
    try {
        error_log("开始发送履约确认：tradeNo={$tradeNo}");
        
        $alipayClient = initAlipayClient();
        $request = new \AlipayAipayAgentFulfillmentConfirmRequest();
        
        $request->setBizContent(json_encode([
            'trade_no' => $tradeNo
        ], JSON_UNESCAPED_UNICODE));
        
        $responseResult = $alipayClient->execute($request);
        $responseApiName = str_replace('.', '_', $request->getApiMethodName()) . '_response';
        $response = $responseResult->$responseApiName;
        
        if (!empty($response->code) && $response->code === '10000') {
            error_log("履约确认成功：tradeNo={$tradeNo}");
        } else {
            error_log("履约确认失败：tradeNo={$tradeNo}, errorCode=" . 
                ($response->sub_code ?? 'Unknown') . ", errorMsg=" . 
                ($response->sub_msg ?? 'Unknown'));
        }
        
    } catch (Exception $e) {
        error_log("履约确认异常：tradeNo={$tradeNo}, error=" . $e->getMessage());
    }
}

// ==================== 路由处理 ====================

// 简单路由
$requestUri = parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH);
if ($requestUri === RESOURCE_CONFIG['path'] && $_SERVER['REQUEST_METHOD'] === 'GET') {
    handleResourceRequest();
} else {
    http_response_code(404);
    echo json_encode(['code' => 'NOT_FOUND', 'message' => '接口不存在'], JSON_UNESCAPED_UNICODE);
}

File: .agents/skills/alipay-payment-integration/references/code-examples/csharp/10_AI收/A2MPaymentDemo.cs
/*
 * A2M 智能收产品接入示例 - C# 版本
 * 
 * 本文件演示完整的智能收产品接入流程：
 * 1. 返回 402 Payment-Needed Header
 * 2. 验证 Payment-Proof 支付凭证
 * 3. 发送履约回执确认
 * 4. 返回资源内容
 * 
 * 依赖安装：
 * dotnet add package AlipaySDKNet.Standard
 * dotnet add package Microsoft.AspNetCore.App
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.DependencyInjection;
using Aop.Api;
using Aop.Api.Request;
using Aop.Api.Response;
using Aop.Api.Domain;

namespace A2MPaymentDemo
{
    class Program
    {
        // ==================== 配置信息（实际使用时请从配置中心读取）====================
        private static readonly AlipayConfig AlipayConfig = new AlipayConfig
        {
            ServerUrl = "https://openapi.alipay.com/gateway.do",
            AppId = "2026000123456789",
            PrivateKey = "MIIEvQIBADANBgkq...", // 请填写您的应用私钥（PKCS#1 格式）
            AlipayPublicKey = "MIIBIjANBgkq...", // 请填写您的支付宝公钥
            Format = "json",
            Charset = "UTF-8",
            SignType = "RSA2"
        };

        private static readonly string SellerId = "2088123456789012"; // 商户 ID（2088 格式）
        private static readonly string ServiceId = "service_ai_content_001"; // 商户服务 ID
        private static readonly string MerchantPrivateKey = "MIIEvQIBADANBgkq..."; // 请填写您的应用私钥（用于商家签名）
        
        private const string ResourcePath = "/demo/a2m/resource";
        private const string GoodsName = "AI 生成内容服务";

        // 初始化支付宝客户端
        private static readonly IAopClient AlipayClient = new DefaultAopClient(AlipayConfig);

        static void Main(string[] args)
        {
            var builder = WebApplication.CreateBuilder(args);
            var app = builder.Build();

            // 路由处理
            app.MapGet(ResourcePath, async (HttpRequest request, HttpResponse response) =>
            {
                // 获取 Payment-Proof Header
                var paymentProof = request.Headers["Payment-Proof"].FirstOrDefault();

                // 场景 1：用户未支付，返回 402 + Payment-Needed Header
                if (string.IsNullOrWhiteSpace(paymentProof))
                {
                    await CreatePaymentRequiredResponse(response);
                    return;
                }

                // 场景 2：用户已支付，验证 Payment-Proof 并返回资源
                await VerifyPaymentAndDeliverResource(response, paymentProof);
            });

            Console.WriteLine($"A2M 智能收服务已启动：http://localhost:5000{ResourcePath}");
            Console.WriteLine("测试步骤：");
            Console.WriteLine($"1. 无 Payment-Proof Header: curl http://localhost:5000{ResourcePath}");
            Console.WriteLine($"2. 有 Payment-Proof Header: curl -H \"Payment-Proof: <value>\" http://localhost:5000{ResourcePath}");

            app.Run();
        }

        // ==================== 工具方法 ====================

        /// <summary>
        /// 格式化支付宝时间戳：yyyy-MM-dd HH:mm:ss
        /// </summary>
        private static string FormatAlipayTimestamp(DateTime? dateTime = null)
        {
            return (dateTime ?? DateTime.Now).ToString("yyyy-MM-dd HH:mm:ss");
        }

        /// <summary>
        /// Base64URL 编码
        /// </summary>
        private static string Base64UrlEncode(string data)
        {
            var bytes = Encoding.UTF8.GetBytes(data);
            var base64 = Convert.ToBase64String(bytes);
            return base64.Replace('+', '-').Replace('/', '_').TrimEnd('=');
        }

        /// <summary>
        /// Base64URL 解码
        /// </summary>
        private static string Base64UrlDecode(string data)
        {
            // 补充 padding
            var padding = data.Length % 4;
            if (padding != 0)
            {
                data += new string('=', 4 - padding);
            }
            
            var base64 = data.Replace('-', '+').Replace('_', '/');
            var bytes = Convert.FromBase64String(base64);
            return Encoding.UTF8.GetString(bytes);
        }

        /// <summary>
        /// 生成商家签名（seller_signature）
        /// </summary>
        private static string GenerateSellerSignature(Dictionary<string, string> parameters, string privateKey)
        {
            // 1. 按 key 字典序排序
            var sortedKeys = parameters.Keys.OrderBy(k => k).ToList();
            
            // 2. 拼接签名内容
            var signContent = new StringBuilder();
            var first = true;
            foreach (var key in sortedKeys)
            {
                var value = parameters[key];
                if (!string.IsNullOrEmpty(value))
                {
                    if (!first)
                    {
                        signContent.Append("&");
                    }
                    signContent.Append($"{key}={value}");
                    first = false;
                }
            }

            // 3. RSA2 签名
            using (var rsa = new RSACryptoServiceProvider())
            {
                rsa.ImportPkcs8PrivateKey(Convert.FromBase64String(privateKey), out _);
                var signature = rsa.SignData(Encoding.UTF8.GetBytes(signContent.ToString()), HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
                return Convert.ToBase64String(signature);
            }
        }

        // ==================== 智能收产品接入示例接口 ====================

        /// <summary>
        /// 创建 402 支付请求响应
        /// </summary>
        private static async Task CreatePaymentRequiredResponse(HttpResponse response)
        {
            try
            {
                // 1. 构造订单信息
                var outTradeNo = $"ORDER_{DateTimeOffset.Now.ToUnixTimeMilliseconds()}";
                var amount = "0.01";
                var currency = "CNY";
                var resourceId = ResourcePath;
                var goodsName = GoodsName;

                // 2. 计算支付截止时间（30 分钟后）
                var payBefore = DateTimeOffset.Now.AddMinutes(30).ToString("o");

                // 3. 生成商家签名
                var signParams = new Dictionary<string, string>
                {
                    ["amount"] = amount,
                    ["currency"] = currency,
                    ["goods_name"] = goodsName,
                    ["out_trade_no"] = outTradeNo,
                    ["pay_before"] = payBefore,
                    ["resource_id"] = resourceId,
                    ["seller_id"] = SellerId,
                    ["service_id"] = ServiceId
                };

                var sellerSignature = GenerateSellerSignature(signParams, MerchantPrivateKey);

                // 4. 构造 Payment-Needed Header 内容
                var paymentNeeded = new
                {
                    protocol = new
                    {
                        out_trade_no = outTradeNo,
                        amount = amount,
                        currency = currency,
                        resource_id = resourceId,
                        pay_before = payBefore,
                        seller_signature = sellerSignature,
                        seller_sign_type = "RSA2",
                        seller_unique_id = SellerId
                    },
                    method = new
                    {
                        seller_name = "测试商户",
                        seller_id = SellerId,
                        seller_app_id = AlipayConfig.AppId,
                        goods_name = goodsName,
                        seller_unique_id_key = "seller_id",
                        service_id = ServiceId
                    }
                };

                // 5. Base64URL 编码
                var paymentNeededJson = JsonSerializer.Serialize(paymentNeeded, new JsonSerializerOptions
                {
                    PropertyNamingPolicy = JsonNamingPolicy.CamelCase
                });
                var paymentNeededEncoded = Base64UrlEncode(paymentNeededJson);

                // 6. 构造 402 响应
                var responseBody = new
                {
                    code = "Payment-Needed",
                    message = "需要支付",
                    out_trade_no = outTradeNo,
                    amount = amount,
                    currency = currency,
                    goods_name = goodsName
                };

                Console.WriteLine($"创建支付订单成功：outTradeNo={outTradeNo}, amount={amount}");

                response.StatusCode = 402;
                response.Headers["Payment-Needed"] = paymentNeededEncoded;
                response.ContentType = "application/json; charset=utf-8";
                await response.WriteAsJsonAsync(responseBody);

            }
            catch (Exception ex)
            {
                Console.WriteLine($"创建订单失败：{ex.Message}");
                response.StatusCode = 500;
                response.ContentType = "application/json; charset=utf-8";
                await response.WriteAsJsonAsync(new
                {
                    code = "CREATE_ORDER_ERROR",
                    message = $"创建订单失败：{ex.Message}"
                });
            }
        }

        /// <summary>
        /// 验证支付凭证并交付资源
        /// </summary>
        private static async Task VerifyPaymentAndDeliverResource(HttpResponse response, string paymentProof)
        {
            try
            {
                // 1. 从 Payment-Proof 中解析订单信息
                string paymentProofValue = null;
                string tradeNo = null;
                string clientSession = null;

                try
                {
                    var decodedProof = Base64UrlDecode(paymentProof);
                    using (var proofJson = JsonDocument.Parse(decodedProof))
                    {
                        var root = proofJson.RootElement;
                        
                        if (root.TryGetProperty("protocol", out var protocol))
                        {
                            paymentProofValue = protocol.GetProperty("payment_proof").GetString();
                            tradeNo = protocol.GetProperty("trade_no").GetString();
                        }

                        if (root.TryGetProperty("method", out var method))
                        {
                            clientSession = method.GetProperty("client_session").GetString();
                        }
                    }

                    // 校验必要字段
                    if (string.IsNullOrWhiteSpace(paymentProofValue))
                    {
                        response.StatusCode = 400;
                        response.ContentType = "application/json; charset=utf-8";
                        await response.WriteAsJsonAsync(new
                        {
                            code = "INVALID_PAYMENT_PROOF_FORMAT",
                            message = "Payment-Proof 格式错误：缺少 payment_proof"
                        });
                        return;
                    }

                    if (string.IsNullOrWhiteSpace(tradeNo))
                    {
                        response.StatusCode = 400;
                        response.ContentType = "application/json; charset=utf-8";
                        await response.WriteAsJsonAsync(new
                        {
                            code = "INVALID_PAYMENT_PROOF_FORMAT",
                            message = "Payment-Proof 格式错误：缺少 trade_no"
                        });
                        return;
                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"Payment-Proof 解析失败：{ex.Message}");
                    response.StatusCode = 400;
                    response.ContentType = "application/json; charset=utf-8";
                    await response.WriteAsJsonAsync(new
                    {
                        code = "INVALID_PAYMENT_PROOF_FORMAT",
                        message = $"Payment-Proof 格式错误：{ex.Message}"
                    });
                    return;
                }

                // 2. 调用支付宝 API 验证支付凭证
                var request = new AlipayAipayAgentPaymentVerifyRequest();
                request.BizContent = JsonSerializer.Serialize(new
                {
                    payment_proof = paymentProofValue,
                    trade_no = tradeNo,
                    client_session = clientSession
                }, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });

                var verifyResponse = await AlipayClient.ExecuteAsync(request);

                // 3. 验证失败，返回错误
                if (verifyResponse.Code != "10000")
                {
                    Console.WriteLine($"支付凭证验证失败：{verifyResponse.SubMsg}");
                    response.StatusCode = 400;
                    response.ContentType = "application/json; charset=utf-8";
                    await response.WriteAsJsonAsync(new
                    {
                        code = verifyResponse.SubCode ?? "VERIFY_FAILED",
                        message = verifyResponse.SubMsg ?? "支付凭证验证失败"
                    });
                    return;
                }

                // 4. 验证成功，获取订单信息
                var verifyTradeNo = verifyResponse.TradeNo;
                var verifyOutTradeNo = verifyResponse.OutTradeNo;
                var resourceId = verifyResponse.ResourceId;
                var active = verifyResponse.Active;

                Console.WriteLine($"支付凭证验证成功：tradeNo={verifyTradeNo}, outTradeNo={verifyOutTradeNo}");

                // 5. 校验凭证有效性（active=true 表示凭证有效）
                if (active != true)
                {
                    Console.WriteLine($"支付凭证无效或已过期：outTradeNo={verifyOutTradeNo}");
                    response.StatusCode = 400;
                    response.ContentType = "application/json; charset=utf-8";
                    await response.WriteAsJsonAsync(new
                    {
                        code = "INVALID_PAYMENT_PROOF",
                        message = "支付凭证无效或已过期"
                    });
                    return;
                }

                // 6. 【TODO】查询订单是否存在（以数据库为准）
                // 7. 【TODO】资源防串校验
                // 8. 【TODO】履约防重放校验

                // 9. 生成资源内容
                var serviceResult = GenerateServiceResource(resourceId);

                // 10. 【TODO】履约记录落库
                // 11. 【TODO】更新订单状态

                Console.WriteLine($"履约成功：outTradeNo={verifyOutTradeNo}, tradeNo={verifyTradeNo}");

                // 12. 发送履约确认到支付宝
                await SendFulfillmentConfirm(verifyTradeNo);

                // 13. 构造 Payment-Validation Header
                var paymentValidation = new
                {
                    trade_no = verifyTradeNo,
                    out_trade_no = verifyOutTradeNo,
                    validated = true,
                    resource_id = resourceId
                };

                var paymentValidationJson = JsonSerializer.Serialize(paymentValidation, new JsonSerializerOptions
                {
                    PropertyNamingPolicy = JsonNamingPolicy.CamelCase
                });
                var paymentValidationEncoded = Base64UrlEncode(paymentValidationJson);

                // 14. 返回资源内容
                var responseBody = new
                {
                    resource_id = resourceId,
                    content = serviceResult,
                    trade_no = verifyTradeNo,
                    out_trade_no = verifyOutTradeNo,
                    already_fulfilled = false
                };

                response.Headers["Payment-Validation"] = paymentValidationEncoded;
                response.ContentType = "application/json; charset=utf-8";
                await response.WriteAsJsonAsync(responseBody);
            }
            catch (Exception ex)
            {
                Console.WriteLine($"支付凭证验证异常：{ex.Message}");
                response.StatusCode = 500;
                response.ContentType = "application/json; charset=utf-8";
                await response.WriteAsJsonAsync(new
                {
                    code = "VERIFY_FAILED",
                    message = $"支付凭证验证失败：{ex.Message}"
                });
            }
        }

        /// <summary>
        /// 生成服务资源内容
        /// </summary>
        private static object GenerateServiceResource(string resourceId)
        {
            return new
            {
                status = "success",
                service_type = "AI_CONTENT_GENERATION",
                resource_id = resourceId,
                content = "这是 AI 生成的内容示例，可根据实际业务替换为任意数字服务内容",
                generated_at = DateTime.Now.ToString("o")
            };
        }

        /// <summary>
        /// 发送履约确认
        /// </summary>
        private static async Task SendFulfillmentConfirm(string tradeNo)
        {
            try
            {
                Console.WriteLine($"开始发送履约确认：tradeNo={tradeNo}");

                var request = new AlipayAipayAgentFulfillmentConfirmRequest();
                request.BizContent = JsonSerializer.Serialize(new
                {
                    trade_no = tradeNo
                }, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });

                var response = await AlipayClient.ExecuteAsync(request);

                if (response.Code == "10000")
                {
                    Console.WriteLine($"履约确认成功：tradeNo={tradeNo}");
                }
                else
                {
                    Console.WriteLine($"履约确认失败：tradeNo={tradeNo}, errorCode={response.SubCode}, errorMsg={response.SubMsg}");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"履约确认异常：tradeNo={tradeNo}, error={ex.Message}");
            }
        }
    }
}
